content/sensu-go/6.8/operations/deploy-sensu/secure-postgres.md

```diff
@@ -0,0 +1,492 @@
+cp /home/user/ca.pem /var/lib/pgsql/14/data/tls/
+chown -R postgres:postgres /var/lib/pgsql/14/data
+{{< /code >}}
+
+   {{< code shell "Ubuntu/Debian">}}
+sudo mkdir /etc/postgresql/14/main/tls
+cd /etc/postgresql/14/main/tls
+cp /home/user/postgres.example.com* /etc/postgresql/14/main/tls/
+cp /home/user/ca.pem /etc/postgresql/14/main/tls/
+chown -R postgres:postgres /etc/postgresql/14/main/
+{{< /code >}}
+
+{{< /language-toggle >}}
+
+3. Open the PostgreSQL configuration file `postgresql.conf` in your code editor and edit the following lines to enable TLS:
+
+   {{< language-toggle >}}
+
+   {{< code shell "RHEL/CentOS">}}
+# vim /var/lib/pgsql/14/data/postgresql.conf
+
+# - SSL -
+
+ssl = on
+ssl_ca_file = '/var/lib/pgsql/14/data/tls/ca.pem'
+ssl_cert_file = '/var/lib/pgsql/14/data/tls/postgres.example.com.pem'
+ssl_key_file = '/var/lib/pgsql/14/data/tls/postgres.example.com-key.pem'
+{{< /code >}}
+
+   {{< code shell "Ubuntu/Debian" >}}
+# vim /etc/postgresql/14/main/postgresql.conf
+
+# - SSL -
+
+ssl = on
+ssl_ca_file = '/etc/postgresql/14/main/tls/ca.pem'
+ssl_cert_file = '/etc/postgresql/14/main/tls/postgres.example.com.pem'
+ssl_key_file = '/etc/postgresql/14/main/tls/postgres.example.com-key.pem'
+{{< /code >}}
+
+   {{< / language-toggle >}}
+
+   Save your changes and close the file.
+
+4. Open the `pg_hba.conf` file in your respective Linux distribution and edit the following lines to configure host-based authentication to accept certificates only when accessing the `sensu_events` database:
+
+   {{< language-toggle >}}
+
+   {{< code shell  "RHEL/CentOS" >}}
+
+# vim /var/lib/pgsql/14/data/pg_hba.conf
+
+# Prevent "postgres" superuser login via a certificate
+hostssl all             postgres        ::/0                    reject
+hostssl all             postgres        0.0.0.0/0               reject
+
+# Rules for Sensu DB connection
+hostssl sensu_events    sensu           0.0.0.0/0               cert
+{{< /code >}}
+
+   {{< code shell  "Ubuntu/Debian" >}}
+
+# vim /etc/postgresql/14/main/pg_hba.conf
+
+# Prevent "postgres" superuser login via a certificate
+hostssl all             postgres        ::/0                    reject
+hostssl all             postgres        0.0.0.0/0               reject
+
+# Rules for Sensu DB connection
+hostssl sensu_events    sensu           0.0.0.0/0               cert
+{{< /code >}}
+
+{{< /language-toggle >}}
+
+5. Restart PostgreSQL:
+
+   {{< language-toggle >}}
+
+   {{< code shell "RHEL/Centos" >}}
+sudo systemctl restart postgresql-14.service
+{{< /code >}}
+
+   {{< code shell "Ubuntu/Debian" >}}
+sudo systemctl restart postgresql.service
+{{< /code >}}
+
+{{< /language-toggle >}}
+
+Now that you've configured PostgreSQL to use TLS and your Sensu user is required to authenticate with a certificate, complete one final step to ensure that PostgreSQL uses the environment variables set earlier in this guide when constructing the PostgreSQL DSN.
```

Comment:
I edited this sentence (trying to remove passive language) but I'm not sure I got it right. Previously, this is what it said:
```
Now that you've configured PostgreSQL to use TLS and your Sensu user is required to authenticate with a certificate, you'll need to complete one final step to ensure that the environment variables set earlier in this guide are used when constructing the PostgreSQL DSN.
```

Reply:
👍 I like it.